pkg/operator/hive/manageddns.go

```diff
@@ -18,36 +18,40 @@ import (
 )
 
 const (
-	managedDomainsConfigMapNamePrefix = "managed-domains-"
-	managedDomainsConfigMapKey        = "managed-domains"
-	configMapLabel                    = "managed-domains"
-	configMapMountPath                = "/data/config"
+	managedDomainsConfigMapName = "hive-managed-domains"
+	managedDomainsConfigMapKey  = "managed-domains"
+	configMapLabel              = "managed-domains"
+	configMapMountPath          = "/data/config"
 )
 
 // configureManagedDomains will create a new configmap holding the managed domains settings (if necessary), or simply
-// return the current configmap of the current deployment if the settings it contains match the desired settings.
-func (r *ReconcileHiveConfig) configureManagedDomains(logger log.FieldLogger, h resource.Helper, instance *hivev1.HiveConfig, namespacesToClean []string) (*corev1.ConfigMap, error) {
+// use the current configmap of the current deployment if the settings it contains match the desired settings. The
+// first return value is a hash (MD5 sum) of the configmap's data, which can be used to indicate whether it changed.
+func (r *ReconcileHiveConfig) configureManagedDomains(logger log.FieldLogger, h resource.Helper, instance *hivev1.HiveConfig, namespacesToClean []string) (string, error) {
 	// Scrub old target namespaces.
 	for _, ns := range namespacesToClean {
 		// Cheat: getCurrentConfigMap deletes any configmap that doesn't have the specified data.
 		// (Technically this could miss a configmap with no data. That's not a realistic problem
 		// as we always create them with managedDomainsConfigMapKey.)
 		_, err := r.getCurrentConfigMap(nil, h, ns, logger)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to scrub managed domains config maps from old target namespace %s", ns)
+			return "", errors.Wrapf(err, "failed to scrub managed domains config maps from old target namespace %s", ns)
 		}
 	}
 
 	domains, err := json.Marshal(instance.Spec.ManagedDomains)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to marshal managed domains list into the configmap")
+		return "", errors.Wrap(err, "failed to marshal managed domains list into the configmap")
 	}
 
 	newConfigMapData := map[string]string{managedDomainsConfigMapKey: string(domains)}
 
+	// TODO: The algorithm from here down can be simplified once we're sure we've flushed out any
```

Comment:
Should we make a Jira to follow up on this in a month or two?

Reply:
I was thinking about that.

I don't think we ever really _know_ we've flushed out the old configmaps, so arguably it's _never_ safe to do this.

However, if we refactor this code accordingly and it gets deployed in such an env (e.g. upgrade from ACM 2.4 to 2.5?) the worst thing that happens is we leave a stale, unused configmap lying around. It doesn't have any sensitive data, so meh.

Thoughts?

Reply:
I'm fine doing a best-effort migration. I forgot about ACM, so perhaps we should put the Jira card out closer to 6 months into the future to make sure ACM picks up a release with this cleanup code.

Reply:
https://issues.redhat.com/browse/HIVE-1700